xilem_web/src/after_update.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use xilem_core::{MessageResult, Mut, View, ViewId, ViewPathTracker};
+
+use crate::{interfaces::Element, DynMessage, ViewCtx};
+
+pub struct AfterUpdate<E, F> {
+    pub(crate) element: E,
+    pub(crate) callback: F,
+}
+
+impl<E, F> AfterUpdate<E, F> {
+    pub fn new(element: E, callback: F) -> AfterUpdate<E, F> {
+        AfterUpdate { element, callback }
+    }
+}
+
+pub struct AfterUpdateState<E, S> {
+    element: E,
+    child_state: S,
+    child_id: ViewId,
+}
+
+impl<State, Action, E, F> View<State, Action, ViewCtx, DynMessage> for AfterUpdate<E, F>
+where
+    State: 'static,
+    Action: 'static,
+    E: Element<State, Action>,
+    E::Element: Clone + PartialEq,
+    F: Fn(&mut State, &E::Element) + 'static,
+{
+    type Element = E::Element;
+
+    type ViewState = AfterUpdateState<Self::Element, E::ViewState>;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let id = ViewId::new(0); // FIXME: what's the right number here?
```

Comment:
This should be fine, our whole view id system has changed to generational indexing (and the generation doesn't change in this view).
Thus views don't have a real id anymore, and are instead "unique" by the view path that is created by `ViewPathTracker::with_id()` while traversing in one of these methods (implemented by `ViewCtx`), but as an id could be changing in the ascendents (e.g. because an `AnyView` has changed its type), which should be kept in mind.

The child view below thus also doesn't need an explicit id anymore as well.